src/backend/distributed/shared_library_init.c

```diff
@@ -1315,6 +1319,23 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_client_connections",
+		gettext_noop("Sets the maximum number of connections regular clients can make"),
+		gettext_noop("To ensure that a Citus cluster has a sufficient number of "
+					 "connection slots to serve queries internally, it can be "
+					 "useful to reserve connection slots for Citus internal "
+					 "connections. When max_client_connections is set to a value "
+					 "below max_connections, the remaining connections are reserved "
+					 "for connections between Citus nodes. This does not affect "
+					 "superuser_reserved_connections. If set to -1, no connections "
+					 "are reserved."),
+		&MaxClientConnections,
```

Comment:
I guess the max cannot exceed MaxConnections in practice, but probably it is too much effort to enforce. So, just wanted to explicitly ask if this observation is true? Maybe 1 more sentence in the comment could be useful, like: `When max_client_connections is set to a value above max_connection, it practically becomes useless as the clients would fail earlier`

Reply:
changed the max to MaxConnections